dali/operators/util/randomizer.cuh

```diff
@@ -64,6 +64,70 @@ struct curand_normal_dist<double> {
   }
 };
 
+
+template <typename T>
+struct curand_uniform_dist {};
+
+template <>
+struct curand_uniform_dist<float> {
```

Comment:
As Michał suggested, you can move the common parts to the same place and specialize only the `operator()`.